index.html

```diff
@@ -1177,119 +1177,269 @@ <h2 id="relationship-to-permissions-policy">
+            </li>
+            <li>[=list/Append=] |task| to |tasks|.
+            </li>
+          </ol>
+        </li>
+        <li>Wait for all <a>tasks</a> in |tasks| to have executed.
+        </li>
+      </ol>
+      <section data-cite="webdriver">
+        <h3 id="automation-webdriver">
+          Automated testing with [[WebDriver]]
         </h3>
-        <table>
-          <tbody>
-            <tr>
-              <th>
-                HTTP Method
-              </th>
-              <th>
-                <a data-lt="extension command URI template">URI Template</a>
-              </th>
-            </tr>
-            <tr>
-              <td>
-                POST
-              </td>
-              <td>
-                /session/{session id}/permissions
-              </td>
-            </tr>
-          </tbody>
-        </table>
         <p>
-          The <dfn class="export" data-dfn-for="extension commands">Set Permission</dfn>
-          <a>extension command</a> simulates user modification of a {{PermissionDescriptor}}'s
-          <a>permission state</a>.
+          This document defines the following <a>extension commands</a> for the [[WebDriver]] specification.
         </p>
+        <section>
+          <h4 id="set-permission-command">
+            Set Permission
+          </h4>
+          <table>
+            <tbody>
+              <tr>
+                <th>
+                  HTTP Method
+                </th>
+                <th>
+                  <a data-lt="extension command URI template">URI Template</a>
+                </th>
+              </tr>
+              <tr>
+                <td>
+                  POST
+                </td>
+                <td>
+                  /session/{session id}/permissions
+                </td>
+              </tr>
+            </tbody>
+          </table>
+          <p>
+            The <dfn class="export" data-dfn-for="extension commands">Set Permission</dfn>
+            <a>extension command</a> simulates user modification of a {{PermissionDescriptor}}'s
+            <a>permission state</a>.
+          </p>
+          <p>
+            The <a>remote end steps</a> are:
+          </p>
+          <ol>
+            <li>Let |parameters| be the |parameters| argument, <a>converted to an IDL value</a> of
+            type {{PermissionSetParameters}}. If this throws an exception, return an [=invalid
+            argument=] [=error=].
+            </li>
+            <li>Let |rootDesc| be |parameters|.{{PermissionSetParameters/descriptor}}.
+            </li>
+            <li>If |parameters|.{{PermissionSetParameters/state}} is an inappropriate <a>permission
+            state</a> for any implementation-defined reason, return a [=invalid argument=] [=error=].
+              <p class="note">
+                For example, <a>user agents</a> that define the "midi" <a>powerful feature</a> as
+                "always on" may choose to reject a command to set the <a>permission state</a> to
```

Comment:
```suggestion
                "always on" can choose to reject a command to set the <a>permission state</a> to
```

How would they do that though? It can only be set to "granted",   "denied",  or "prompt" ?

Reply:
Note: This is part of pre-existing prose. Updated anyway.